tests/ui/autodiff/f64.stderr

```diff
@@ -0,0 +1,40 @@
+updating analysis of val:   %3 = fmul double %2, %2, !dbg !17231 current: {[-1]:Float@double} new {[-1]:Float@double} from   %3 = fmul double %2, %2, !dbg !17231 Changed=0 legal=1
+updating analysis of val: ptr %0 current: {[-1]:Pointer, [-1,0]:Float@double} new {[-1]:Pointer} from   %2 = load double, ptr %0, align 8, !dbg !17230, !noundef !14 Changed=0 legal=1
+updating analysis of val:   %2 = load double, ptr %0, align 8, !dbg !17230, !noundef !14 current: {} new {[-1]:Float@double} from   %2 = load double, ptr %0, align 8, !dbg !17230, !noundef !14 Changed=1 legal=1
+updating analysis of val:   %2 = load double, ptr %0, align 8, !dbg !17230, !noundef !14 current: {[-1]:Float@double} new {[-1]:Float@double} from   %3 = fmul double %2, %2, !dbg !17231 Changed=0 legal=1
+updating analysis of val:   %2 = load double, ptr %0, align 8, !dbg !17230, !noundef !14 current: {[-1]:Float@double} new {[-1]:Float@double} from   %3 = fmul double %2, %2, !dbg !17231 Changed=0 legal=1
+updating analysis of val:   %3 = fmul double %2, %2, !dbg !17231 current: {[-1]:Float@double} new {[-1]:Float@double} from   %3 = fmul double %2, %2, !dbg !17231 Changed=0 legal=1
+analyzing function square
+ + knowndata: ptr %0 : {[-1]:Pointer} - {}
+ + retdata: {[-1]:Float@double}
+updating analysis of val: ptr %0 current: {} new {[-1]:Pointer} from ptr %0 Changed=1 legal=1
+updating analysis of val: ptr %0 current: {[-1]:Pointer} new {[-1]:Pointer} from ptr %0 Changed=0 legal=1
+updating analysis of val:   %3 = fmul double %2, %2, !dbg !385 current: {} new {[-1]:Float@double} from   %3 = fmul double %2, %2, !dbg !385 Changed=1 legal=1
+updating analysis of val:   %3 = fmul double %2, %2, !dbg !385 current: {[-1]:Float@double} new {[-1]:Float@double} from   %3 = fmul double %2, %2, !dbg !385 Changed=0 legal=1
+updating analysis of val: ptr %0 current: {[-1]:Pointer} new {[-1]:Pointer} from   %2 = load double, ptr %0, align 8, !dbg !384, !noundef !14 Changed=0 legal=1
+updating analysis of val:   %2 = load double, ptr %0, align 8, !dbg !384, !noundef !14 current: {} new {} from   %2 = load double, ptr %0, align 8, !dbg !384, !noundef !14 Changed=0 legal=1
+updating analysis of val:   %2 = load double, ptr %0, align 8, !dbg !384, !noundef !14 current: {} new {[-1]:Float@double} from   %3 = fmul double %2, %2, !dbg !385 Changed=1 legal=1
+updating analysis of val:   %2 = load double, ptr %0, align 8, !dbg !384, !noundef !14 current: {[-1]:Float@double} new {[-1]:Float@double} from   %3 = fmul double %2, %2, !dbg !385 Changed=0 legal=1
+updating analysis of val:   %3 = fmul double %2, %2, !dbg !385 current: {[-1]:Float@double} new {[-1]:Float@double} from   %3 = fmul double %2, %2, !dbg !385 Changed=0 legal=1
+updating analysis of val: ptr %0 current: {[-1]:Pointer} new {[-1]:Pointer, [-1,0]:Float@double} from   %2 = load double, ptr %0, align 8, !dbg !384, !noundef !14 Changed=1 legal=1
```

Comment:
I'd assume that the dbg/noundef numbers like 384, 385 and 14 will change soon enough. Best case you replace all fixed constants like 384 and 385 with different variables, there's probably a good regex for it. I.e. on the first use of 384 you assign a name to it, and on all following usages of 384 you assure that it's the same name (value) showing up. There should be examples in Enzyme and the LLVM lit docs on how to do that.
Otherwise at least make sure that all of these two numbers are replaced with an anonymous placeholder, so if they change tests don't break.

Reply:
I'm not sure how you would meaningfully normalize these. Maybe instead, use FileCheck inside a run-make test?

Reply:
That normalization can still be fragile, because can't the instruction ordering also change?

Reply:
We have "run arbitrary code to execute the test" tests, called "run-make" because they used to be makefiles, but they actually are Rust executables now: https://github.com/rust-lang/rust/blob/master/tests/run-make/README.md

You can use that to run filecheck:

https://github.com/rust-lang/rust/blob/4e973370053a5fe87ee96d43c506623e9bd1eb9d/src/tools/run-make-support/src/external_deps/llvm.rs#L23

Reply:
Yeah, I think using FileCheck within a run-make test is going to be least fragile, because you can also "don't care" about attributes and whatever and only match on stuff you do care about.